torch/csrc/autograd/variable.cpp

```diff
@@ -21,6 +24,23 @@ Variable make_variable(at::Tensor data, std::shared_ptr<Function> grad_fn) {
   return make_variable(std::move(data), output_nr, std::move(grad_fn));
 }
 
+FunctionPort Variable::gradient_port() const {
+  // If grad_fn is null (as is the case for a leaf node), we instead
+  // interpret the gradient function to be a grad accumulator,
+  // which will accumulate its inputs into the grad property of the
+  // variable. These nodes get suppressed in some situations,
+  // see "suppress grad accumulation" below. Note that only variables which
+  // have requires_grad=True can have grad accumulators.
+  if (defined()) {
+    if (auto gradient = grad_fn()) {
+      return FunctionPort(gradient, output_nr());
+    } else if (requires_grad()) {
+      return FunctionPort(grad_accumulator());
```

Comment:
This is being used in a few places to replace code that does fewer checks (in particular I think that code assumes requires_grad() and defined()). The autograd is a perf-critical pathway, and this function will be called a lot.  Do we think these extra checks are going to matter?

Reply:
I think we can assume having `requires_grad()`. IIRC `grad_accumulator()` will check it anyway

Reply:
I moved the `defined()` check into `get_next_functions` in `autograd/function.h`, the only place where this check was previously made, and removed this check in `gradient_edge()`, along with the `requires_grad()` check.